NextcloudTalk/SettingsTableViewController.swift

```diff
@@ -0,0 +1,696 @@
+                if error != nil {
+                    if let phoneNumber = phoneNumber {
+                        self.presentPhoneNumberErrorDialog(phoneNumber: phoneNumber)
+                    }
+                    print("Error setting phone number ", error ?? "")
+                } else {
+                    self.view.makeToast(NSLocalizedString("Phone number set successfully", comment: ""), duration: 3, position: CSToastPositionCenter)
+                }
+                self.refreshUserProfile()
+            }
+        })
+        if let setPhoneAction = setPhoneAction {
+            setPhoneAction.isEnabled = false
+            setPhoneNumberDialog.addAction(setPhoneAction)
+        }
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Skip", comment: ""), style: .default) { _ in
+            self.refreshUserProfile()
+        }
+        setPhoneNumberDialog.addAction(cancelAction)
+        self.present(setPhoneNumberDialog, animated: true, completion: nil)
+    }
+
+    func presentPhoneNumberErrorDialog(phoneNumber: String) {
+        var failedPhoneNumberDialog: UIAlertController?
+        var failedPhoneNumber: NBPhoneNumber?
+
+        failedPhoneNumber = try? phoneUtil.parse(phoneNumber, defaultRegion: nil)
+
+        let alertTitle = NSLocalizedString("Could not set phone number", comment: "")
+        var alertMessage = ""
+        if let failedPhoneNumber = failedPhoneNumber {
+            alertMessage = (NSLocalizedString("An error occurred while setting \(failedPhoneNumber) as phone number", comment: ""))
+        }
+        if let failedNumberFormated = try? phoneUtil.format(failedPhoneNumber, numberFormat: NBEPhoneNumberFormat.INTERNATIONAL) {
+            failedPhoneNumberDialog = UIAlertController(
+                       title: alertTitle,
+                       message: alertMessage + "\(failedNumberFormated)",
+                       preferredStyle: .alert)
+        }
+        let retryAction = UIAlertAction(title: NSLocalizedString("Retry", comment: ""), style: .default) { _ in
+            self.presentSetPhoneNumberDialog()
+        }
+        failedPhoneNumberDialog!.addAction(retryAction)
+
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Cancel", comment: ""), style: .default, handler: nil)
+        failedPhoneNumberDialog!.addAction(cancelAction)
+
+        self.present(failedPhoneNumberDialog!, animated: true, completion: nil)
+
+    }
+
+    // MARK: UITextField delegate
+
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        if textField.tag == kPhoneTextFieldTag {
+            let inputPhoneNumber = (textField.text as NSString?)?.replacingCharacters(in: range, with: string)
+            var phoneNumber: NBPhoneNumber?
+            phoneNumber = try? phoneUtil.parse(inputPhoneNumber, defaultRegion: nil)
+            setPhoneAction!.isEnabled = phoneUtil.isValidNumber(phoneNumber)
```

Comment:
```suggestion
            setPhoneAction?.isEnabled = phoneUtil.isValidNumber(phoneNumber)
```
Optional chaining is an easy way to avoid runtime crashes. Unless you need to make sure `setPhoneAction` actually exists here, in that case you should check for that and handle the error case.